src/Component.js

```diff
@@ -1,274 +1,122 @@
-import * as Enum from './Enum.js'
 import * as Utility from './Utility.js'
 import * as Element from './Element.js'
-import * as Assert from './Assert.js'
-import * as Lifecycle from './Lifecycle.js'
 import * as Exception from './Exception.js'
-import * as Node from './Node.js'
 import * as Reconcile from './Reconcile.js'
 import * as Schedule from './Schedule.js'
-import * as Registry from './Registry.js'
 
 /**
- * @type {object}
+ * @param {object} prev
+ * @param {object} next
+ * @return {boolean}
  */
-export var descriptors = {
-	/**
-	 * @param {object} state
-	 * @param {function} callback
-	 * @return {object}
-	 */
-	setState: {
-		value: function (state, callback) {
-			return dispatch(Registry.get(this), this, state, callback)
+export function compare (prev, next) {
+	if (prev !== next) {
+		for (var key in prev) {
+			if (!Utility.has(next, key)) {
```

Comment:
Is it really faster to check key existence before equality? Maybe comparing `Object.keys().length` could be faster

Reply:
Last i checked. Will need to bench to see if that's still true. Intuitively since Object.keys returns an object(array) it seemed like it might not be as good as the battle tested "for in" syntax when the number of keys are few(less than 10) and the two objects are POJO's, which is the common case.